addOns/foxhound/src/main/java/org/zaproxy/zap/extension/foxhound/FoxhoundExportServer.java

```diff
@@ -0,0 +1,119 @@
+package org.zaproxy.zap.extension.foxhound;
+
+import net.htmlparser.jericho.Source;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.zaproxy.addon.network.ExtensionNetwork;
+import org.zaproxy.addon.network.server.HttpMessageHandler;
+import org.zaproxy.addon.network.server.HttpMessageHandlerContext;
+import org.zaproxy.addon.network.server.Server;
+import org.zaproxy.zap.extension.foxhound.config.FoxhoundOptions;
+import org.zaproxy.zap.extension.foxhound.db.TaintInfoStore;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+public class FoxhoundExportServer extends PluginPassiveScanner {
+    private static final Logger LOGGER = LogManager.getLogger(FoxhoundExportServer.class);
+
+    static Server server = null;
+    private static int port = -1;
+
+    private ExtensionNetwork extensionNetwork = null;
+    private TaintInfoStore store = null;
+
+
+    public FoxhoundExportServer() {
+
+    }
+
+    public void start(ExtensionNetwork network, FoxhoundOptions options, TaintInfoStore store) {
+        LOGGER.info("start");
+        this.extensionNetwork = network;
+        this.store = store;
+        port = options.getServerPort();
+        getServer();
+    }
+
+    public void stop() {
+        if (server != null) {
+            try {
+                server.stop();
+            } catch (IOException e) {
+                LOGGER.debug("An error occurred while stopping the proxy.", e);
+            }
+        }
+    }
+
+    private void analyseTaintFlow(String body) {
+        store.deserializeAndAddTaintInfo(body);
+    }
+
+    private Server getServer() {
+        if (server == null) {
+            server = extensionNetwork.createHttpServer(
+                new HttpMessageHandler() {
+                    @Override
+                    public void handleMessage(HttpMessageHandlerContext ctx, HttpMessage msg) {
+                        msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/html");
+                        try {
+                            // For some reason we get the message twice...
```

Comment:
The handler is called twice, message inbound and outbound. Since this is acting as a server, do `ctx.overridden();` to indicate the handler is providing the response.

Reply:
Thanks for the clarification!